db/attachment.go

```diff
@@ -149,7 +153,7 @@ func (db *Database) retrieveAncestorAttachments(doc *document, parentRev string,
 			}
 		}
 	}
-	return parentAttachments
+	return parentAttachments, nil
```

Comment:
This method signature change doesn't look necessary any more (along with the error checking changes above).